pip/req/req_set.py

```diff
@@ -123,6 +133,68 @@ def prep_for_dist(self):
         self.req_to_install.assert_source_matches_version()
 
 
+class SetupCfgDistribution(pkg_resources.Distribution):
+
+    def __init__(self, cfg, location=None):
+        self._cfg = cfg
+        project_name = cfg.get('metadata', 'name')
+        super(SetupCfgDistribution, self).__init__(
+            location=location, project_name=project_name)
+
+    @property
+    def _dep_map(self):
+        try:
+            return self.__dep_map
+        except AttributeError:
+            # requires
+            try:
+                requires = self._cfg.get('metadata', 'requires-dist')
+            except configparser.NoOptionError:
+                requires = []
+            extras = (
+                self._cfg.has_section('extras')
+                and self._cfg.items('extras')) or []
+            dm = {}
+            dm.setdefault(None, []).extend(
+                pkg_resources.parse_requirements(requires))
+            for extra, extra_reqs in extras:
+                dm.setdefault(extra, []).extend(
+                    pkg_resources.parse_requirements(extra_reqs))
+            return dm
+
+    def setup_requires(self):
+        try:
+            setup_requires = self._cfg.get('metadata', 'requires-setup')
```

Comment:
just to repeat what I said in IRC,  I think:

"requires-setup" -->  "setup-requires"
"requires-setup" --> "install-requires"

 "requires-setup" is confusing cause it reverses the causality (i.e. setup requires these things, not the reverse)... and "requires-dist" just creates something that you have to mentally associate with "install_requires", and worse, it almost looks like it might be from PEP426... and we don't want that confusion